### PR TITLE
Fix multichain scripts resume functionality

### PR DESCRIPTION
**Pull request by klkvr**
## Motivation

#4892 removed RPCs from general broadcast data and moved them to ./cache. However, for multichain sequences, RPC endpoints weren't saved anywhere which made resuming multichain scripts impossible

## Solution

Save RPC endpoints used by multichain sequences to ./cache.

Ref #5047

### CODE REVIEW

**Comment by mds1** [comment]
Haven't reviewed code carefully but agree we should fix this so multichain resume works

File: crates/forge/bin/cmd/script/multi.rs [hunk 1]
```diff
@@ -22,10 +22,27 @@ use std::{
...
+pub struct SensitiveMultiChainSequence {
+    pub deployments: Vec<SensitiveScriptSequence>,
+}
+
+impl From<&mut MultiChainSequence> for SensitiveMultiChainSequence {
+    fn from(sequence: &mut MultiChainSequence) -> Self {
+        SensitiveMultiChainSequence {
+            deployments: sequence.deployments.iter_mut().map(|sequence| sequence.into()).collect(),
+        }
+    }
```
[the remark this answers is not in the document]
Comment (mattsse) [reply crates/forge/bin/cmd/script/multi.rs:43]: yeah exactly!

**Review by mattsse — COMMENTED** [review]
sorry, was offline a few days.

last pending nit re function, otherwise lgtm

**Review by Evalir — APPROVED** [review]
lgtm

**Pull request merged**